### DISCUSSION TITLE
Question answering over a PDF document

### DISCUSSION THREAD

**Discussion by clstaudt** [langflow-ai/langflow#243] [Q&A]
Let's try and develop a flow that does Q&A over a PDF document really well.

**Comment by clstaudt** [comment]
With this flow I was able to get some answers that were informed by the PDF content. However, most answers were inaccurate. 

![Screenshot 2023-05-03 at 22 19 50](https://user-images.githubusercontent.com/875194/236040827-5eb69751-ce13-47d4-9a7c-7fea746da4f7.png)

Which modifications could lead to better performance?

**Reply by ogabrielluiz** [reply]
Did you try toning down the Temperature? 

What I find with pdfs is that sometimes they don't work because the file isn't really a PDF, just a few images saved into a PDF.

**Reply by clstaudt** [reply]
Temperature: What effect would you expect?

The model is aware of the content of the PDF to some extent, so I assume the text has been extracted correctly.

**Reply by clstaudt** [reply]
I set the temperature to 0.1, but the answers are still quite poor, sometimes unrelated to the input PDF.

![Screenshot 2023-05-03 at 23 14 51](https://user-images.githubusercontent.com/875194/236051554-01e7bba0-e119-44a7-a8fc-9556e897e397.png)

**Reply by ogabrielluiz** [reply]
One other important aspect might be the `chunk_size`. The number of documents the model requires to find a good answer is more than what it can receive with chunks of this size.

The default number of documents is 4 but we can create an issue to make that a parameter too.

**Reply by clstaudt** [reply]
@ogabrielluiz Could you elaborate on that please?

- documents = chunks?
- Where does a default number of documents enter the workflow?
- Can we display diagnostic info for the text splitting, like how many chunks the pdf text gets split into?

**Comment by mauriled** [comment]
responses are quite poor, maybe with a promptemplate?

**Comment by Cristhianzl** [comment]
Hi! 👋

We are using other channels as our official means of communication with users. We apologize for the delayed response.
For faster assistance, please direct your questions to our Discord channel or open a GitHub issue, as we are much more active on these platforms.

[Langflow Discord Server](https://discord.com/invite/EqksyE2EX9)

Thank you for your understanding.

Best regards,
The Langflow Team
